pypowsybl/perunitnetwork.py

```diff
@@ -0,0 +1,636 @@
+        lines['i1'] /= self.sn * 10 ** 3 / (self.sqrt3 * lines['nominal_v'])
+        lines['i2'] /= self.sn * 10 ** 3 / (self.sqrt3 * lines['nominal_v'])
+        lines['r'] /= lines['nominal_v'] ** 2 / self.sn
+        lines['x'] /= lines['nominal_v'] ** 2 / self.sn
+        lines['g1'] /= self.sn / lines['nominal_v'] ** 2
+        lines['g2'] /= self.sn / lines['nominal_v'] ** 2
+        lines['b1'] /= self.sn / lines['nominal_v'] ** 2
+        lines['b2'] /= self.sn / lines['nominal_v'] ** 2
+        return lines.drop('nominal_v', axis=1)
+
+    def get_2_windings_transformers(self) -> _pd.DataFrame:
+        """ Get 2 windings transformers as a ``Pandas`` data frame.
+
+        Returns:
+            a 2 windings transformers data frame
+        """
+        two_windings_transformers = self.get_elements(_pypowsybl.ElementType.TWO_WINDINGS_TRANSFORMER)
+        two_windings_transformers['p1'] /= self.sn
+        two_windings_transformers['q1'] /= self.sn
+        two_windings_transformers['p2'] /= self.sn
+        two_windings_transformers['q2'] /= self.sn
+        two_windings_transformers = _pd.merge(two_windings_transformers, self.get_voltage_levels()['nominal_v'],
+                                              left_on='voltage_level2_id', right_index=True)
+        two_windings_transformers['i1'] /= self.sn * 10 ** 3 / (
+                self.sqrt3 * two_windings_transformers['nominal_v'])
+        two_windings_transformers['i2'] /= self.sn * 10 ** 3 / (
+                self.sqrt3 * two_windings_transformers['nominal_v'])
+        two_windings_transformers['r'] /= two_windings_transformers['nominal_v'] ** 2 / self.sn
+        two_windings_transformers['x'] /= two_windings_transformers['nominal_v'] ** 2 / self.sn
+        two_windings_transformers['g'] /= self.sn / two_windings_transformers['nominal_v'] ** 2
+        two_windings_transformers['b'] /= self.sn / two_windings_transformers['nominal_v'] ** 2
+        two_windings_transformers['rated_u1'] /= two_windings_transformers['nominal_v']
+        two_windings_transformers['rated_u2'] /= two_windings_transformers['nominal_v']
+        return two_windings_transformers.drop(['nominal_v'], axis=1)
+
+    def get_3_windings_transformers(self) -> _pd.DataFrame:
+        """ Get 3 windings transformers as a ``Pandas`` data frame.
+
+        Returns:
+            a 3 windings transformers data frame
+        """
+        three_windings_transformers = self.get_elements(_pypowsybl.ElementType.THREE_WINDINGS_TRANSFORMER)
+        three_windings_transformers['p1'] /= self.sn
+        three_windings_transformers['q1'] /= self.sn
+        three_windings_transformers['p2'] /= self.sn
+        three_windings_transformers['q2'] /= self.sn
+        three_windings_transformers['p3'] /= self.sn
+        three_windings_transformers['q3'] /= self.sn
+        three_windings_transformers = _pd.merge(three_windings_transformers, self.get_voltage_levels()['nominal_v'],
```

Comment:
For each leg, you have to choose the nominal voltage of the side 2 of the leg itself. In our modeling of three windings transformer, for all legs, it is the bus star nominal voltage. So here you have to per-unit with the nominal voltage of the bus star which is indeed the ratedU0 of the three windings transformer. @sylvlecl do you agree ?

Reply:
Yes.
Note that the use of rate_u0 is somewhat arbitrary and "uncommon" compared to everywhere else where we use nominal voltages of the network for per-uniting, so we should mention it in the docstring.

Reply:
It is just that we have decided to put star bus at a nominal voltage equals to ratedU0. And yes, it is arbitrary because this bus is fictitious indeed.